test/test_payu.py

```diff
@@ -301,3 +301,116 @@ def test_list_archive_dirs():
     # Clean up test archive
     shutil.rmtree(tmp_archive)
     shutil.rmtree(tmp_archive_2)
+
+
+@pytest.fixture
+def script_control_dir():
+    # Create a temporary control directory
+    control_dir = tmpdir / 'script_control_dir'
+    control_dir.mkdir()
+
+    yield control_dir
+
+    # Tear down
+    shutil.rmtree(control_dir)
+
+
+def test_run_userscript_python_script(script_control_dir: Path):
+    # Create a simple python script
+    python_script = script_control_dir / 'test_script.py'
+    with open(python_script, 'w') as f:
+        f.writelines([
+            f"with open('{script_control_dir}/output.txt', 'w') as f:\n",
```

Comment:
Is there any need to test the chance of a script with a python shebang "header"?

Reply:
Yeah, I've just added in a new test for that